### PR TITLE
T1.3 save python2 backup file on migration

### PR DESCRIPTION
**Pull request by SHKnudsen**
### Purpose

this PR add mechanism to save a `Python2` backup of the graph.
The backup will only be saved whenever there isnt already a `Python2` backup with the same name it the Dynamo backup folder.

T1 3-SavePython2Backup

### CODE REVIEW

File: src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs [hunk 1]
```diff
@@ -10,11 +12,13 @@ internal class PythonMigrationAssistantViewModel
         public string OldCode { get; set; }
         public string NewCode { get; set; }
         private PythonNode PythonNode { get; set; }
+        public DynamoViewModel DynamoViewModel { get; set; }
 
-        public PythonMigrationAssistantViewModel(PythonNode pythonNode)
+        public PythonMigrationAssistantViewModel(PythonNode pythonNode, DynamoViewModel dynamoViewModel)
```
Comment (radumg) [inline src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs:17]: can we depend on something less hard to get than the `DynamoViewModel` ? Extension access to that is going away eventually.

I think there's a way to get the `backup directory` path without the full viewmode, there's some static settings providers or things like that from what i remember.

File: src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs [hunk 2]
```diff
@@ -25,7 +29,24 @@ private void MigrateCode()
...
+                return;
+
+            DynamoViewModel.SaveAs(path, true);
+        }
+
+        private string GetPythonMigrationBackupPath()
+        {
+            var workspaceName = DynamoViewModel.CurrentSpace.Name;
+            var backupDirectory = DynamoViewModel.Model.PathManager.BackupDirectory;
+            return Path.Combine(backupDirectory, workspaceName) + string.Concat(".", Properties.Resources.PythonMigrationBackupExtension, ".dyn");
```
Comment (radumg) [inline src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs]: i'd add a `var fileName = ...` on the line above to make it more readable

File: src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs (hunk 1 above)
Comment (SHKnudsen) [reply src/PythonMigrationViewExtension/PythonMigrationAssistantViewModel.cs:17]: as far as i can see i can only get it from the `PathManager` which i can only get from the `DynamoModel`. I havent been able to find any other place to get it from, but can have another look.

Im also using the `SaveAs` on the `DynamoModel` which is the same as what Dynamo uses to save the regular backups.

**Comment by alvpickmans** [comment]
Should not the Python Engine change after migration is done? Currently on Gif still shows as `IronPython2`

**Comment by SHKnudsen** [comment]
Yea true @alvpickmans we did actually discuss that, i forgot to add that in. Will add that to the `T1.1` branch.

**Pull request merged**